department-of-veterans-affairs/va.gov-cms
# VACMS 3223:  User Role dimension for the CMS Google Analytics data

## Description

See #3223.

I do not currently have access to Google Tag Manager, so I can't add the necessary mapping.  

I'm setting `currentUserRoles` to an array of user role machine names, e.g.

```
    | role                | roles                                   |
    | "anonymous"         | '["anonymous"]'                         |
    | "authenticated"     | '["authenticated"]'                     | 
    | "content_admin"     | '["authenticated","content_admin"]'     | 
    | "content_publisher" | '["authenticated","content_publisher"]' |
    | "administrator"     | '["authenticated","administrator"]'     |
```

I also added a Behat test and some code to `\CustomDrupal\FeatureContext` to support the test and any other tests we want to perform on the JS Drupal Settings or on the GTM dataLayer.  Please let me know if this is inappropriate or should've been done in a different place.

## Testing done
I added a Behat test and some supporting code and ran it for the roles listed above.

## Screenshots
<img width="554" alt="Screen Shot 2020-10-26 at 12 51 51 PM" src="https://user-images.githubusercontent.com/1318579/97209220-0f076280-178a-11eb-86a3-d3fd439b8c7b.png">

## QA steps
As any user
1. Log in and view any page in Chrome or some other browser with a developer console.
2. Open developer console and type `dataLayer[1].currentUserRoles`.
3. Verify that the console responds with a comma-separated list of user roles.

## Definition of Done
- [x] Mapping needs to be added in GTM to set the appropriate GA custom dimension.  I have access to GA but not GTM yet.
